rslib/src/search/sqlwriter.rs

```diff
@@ -214,14 +214,21 @@ impl SqlWriter<'_> {
         Ok(())
     }
 
-    fn write_rated(&mut self, days: u32, ease: &EaseKind) -> Result<()> {
+    fn write_rated(&mut self, op: &str, days: i64, ease: &EaseKind) -> Result<()> {
         let today_cutoff = self.col.timing_today()?.next_day_at;
-        let target_cutoff_ms = (today_cutoff - 86_400 * i64::from(days)) * 1_000;
-        write!(
-            self.sql,
-            "c.id in (select cid from revlog where id>{}",
-            target_cutoff_ms,
-        )
+        let target_cutoff_ms = (today_cutoff + 86_400 * days) * 1_000;
+        let day_before_cutoff_ms = (today_cutoff + 86_400 * (days - 1)) * 1_000;
+
+        write!(self.sql, "c.id in (select cid from revlog where id").unwrap();
+
+        match op {
+            ">" => write!(self.sql, " >= {}", target_cutoff_ms),
+            ">=" => write!(self.sql, " >= {}", day_before_cutoff_ms),
+            "<" => write!(self.sql, " < {}", day_before_cutoff_ms),
+            "<=" => write!(self.sql, " < {}", target_cutoff_ms),
+            "=" => write!(self.sql, " between {} and {}", day_before_cutoff_ms, target_cutoff_ms - 1),
+            _ /* "!=" */ => write!(self.sql, " not between {} and {}", day_before_cutoff_ms, target_cutoff_ms - 1),
```

Comment:
Minor nit: matching '!='explicitly, and mapping _ => unreachable!("unexpected op") would ensure we don't accidentally fall through if another comparator is ever added in the future.